flash: Fixes on stm32f4 driver

erwango opened the pull request:
commit 5200669
```
Author: Erwan Gouriou <erwan.gouriou@linaro.org>
Date:   Thu Oct 24 15:10:38 2019 +0200

    drivers/flash: stm32f4: Return error on Dual Bank configuration
    
    Some stm32f4 1MB SoCs support optional Dual Bank configuration.
    This is not yet supported by stm32f4 driver, so report an
    error when configuration is detected
    
    Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
```

commit 543b3cc
```
Author: Erwan Gouriou <erwan.gouriou@linaro.org>
Date:   Thu Oct 24 15:15:53 2019 +0200

    drivers/flash: stm32f4: Offset SNB from sector 12 on 2MB parts
    
    On 2MB parts, on sector 12 and above SNB is offset by 4.
    
    Fixes #20016
    
    Signed-off-by: Florian Vaussard <florian.vaussard@gmail.com>
    Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
```

erwango committed:
drivers/flash: stm32f4: Offset SNB from sector 12 on 2MB parts

On 2MB parts, on sector 12 and above SNB is offset by 4.

Fixes #20016

Signed-off-by: Florian Vaussard <florian.vaussard@gmail.com>
Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

MaureenHelm commented on drivers/flash/flash_stm32f4x.c:
```diff
@@ -20,6 +20,17 @@ bool flash_stm32_valid_range(struct device *dev, off_t offset, u32_t len,
+	 */
+	if (regs->optcr & FLASH_OPTCR_DB1M) {
+		/* Device configured in Dual Bank, but not supported for now */
+		return -ENOTSUP;
```
The function prototype says you should return a bool, not an int

MaureenHelm replied:
Changing this to -1 is still returning an int

erwango replied:
I should not hack during meetings... Thanks

erwango committed:
drivers/flash: stm32f4: Return error on Dual Bank configuration

Some stm32f4 1MB SoCs support optional Dual Bank configuration.
This is not yet supported by stm32f4 driver, so report an
error when configuration is detected

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
